testjson/format.go

```diff
@@ -73,6 +75,55 @@ func standardJSONFormat(out io.Writer) EventFormatter {
 	})
 }
 
+func testDoxFormat(out io.Writer, opts FormatOptions) EventFormatter {
+	buf := bufio.NewWriter(out)
+	type Result struct {
+		Event    TestEvent
+		Sentence string
+	}
+	getIcon := icon
+	if opts.UseHiVisibilityIcons {
+		getIcon = iconHiVis
+	}
+	results := map[string][]Result{}
+	return eventFormatterFunc(func(event TestEvent, exec *Execution) error {
+		switch {
+		case event.PackageEvent():
+			if !event.Action.IsTerminal() {
+				return nil
+			}
+			if len(results[event.Package]) == 0 {
+				// No testdox for you
+				return nil
+			}
+			fmt.Fprintf(buf, "%s:\n", event.Package)
+			tests := results[event.Package]
+			sort.Slice(tests, func(i, j int) bool {
+				return tests[i].Sentence < tests[j].Sentence
+			})
+			for _, r := range tests {
+				fmt.Fprintf(buf, " %s %s %s\n",
+					getIcon(r.Event.Action),
+					r.Sentence,
+					r.Event.ElapsedFormatted())
+			}
+			fmt.Fprintln(buf)
+			return buf.Flush()
+		case event.Action.IsTerminal():
```

Comment:
I changed this to `IsTerminal` so that skipped tests show up in the output as well. Does that sound ok to you?

Reply:
Again, it makes sense to show skipped tests in a test runner, but a documentation generator takes the opposite view. Since these tests are skipped, they don't by definition assert anything about the behaviour of the code under test.